src/main.go

```diff
@@ -64,6 +64,63 @@ var (
 	pathMonitoring = fmt.Sprintf("%s/monitoring", storageDir)
 )
 
+// EXPERIMENTAL (ref: https://github.com/bench-routes/bench-routes/issues/242)
+
+func GetTimeStampCalc(sub int) string {
+	t := time.Now().Add(time.Second * time.Duration(-sub)) // getting the timestamp `sub` seconds previous
+
+	return s(t.Year()) + "|" + s(t.Month()) + "|" + s(t.Day()) + "|" + s(t.Hour()) + "|" +
+		s(t.Minute()) + "|" + s(t.Second())
+}
+
+func s(st interface{}) string {
+	return fmt.Sprintf("%v", st)
+}
+
+func addNullBlock(chain *tsdb.Chain, path, nullEncodedData, metricType string) {
+	chainReadOnly := tsdb.ReadOnly(path).Refresh() // getting the chain from the disk
+	bstream := chainReadOnly.BlockStream()
+
+	blockStream := *bstream
+
+	lengthBlockStream := len(blockStream)
+
+	if lengthBlockStream == 0 {
+		return
+	}
+
+	block := blockStream[lengthBlockStream-1] // getting the most recent block added to the disk
+
+	blockNormTime := block.GetNormalizedTime()
+	curNormTime := tsdb.GetNormalizedTimeCalc()
+
+	var blockChain []tsdb.Block // for null blocks
+	sub := 0
+	time := int64(processCollectionScrapeTime) // time is in nano seconds
+
+	/*
+		Starting from the current normalized time, we go back
+		and add null blocks until we get past the block
+		normalized time which is the most recent block added
+		at the time when bench-routes was stopped
+	*/
+	for curNormTime > blockNormTime {
+		b := tsdb.GetNewBlock(metricType, nullEncodedData)
+		b.NormalizedTime = curNormTime
+		b.Timestamp = GetTimeStampCalc(sub)
+		blockChain = append(blockChain, *b)
+		sub = sub + int(time/1000000000)
+		curNormTime -= time
+	}
```

Comment:
I do not understand why do we need to add so many 0 blocks?

Some points to note:
1. A zero block is not a null block. Doing `0|0|0|0|0|0|0|0|0|0|0|0|0|0` is not the solution. I like the idea, but that will be far too expensive on memory and in HTTP requests. Imagine a chain which was written last month, and so the blockNormTime will be 1 month behind, then you can imagine the number of blocks that will be added. This will not be scalable and be actually too expensive.
2. For making graphs null, you should first see how will the graphs respect this efficiently, Fortunately, apexcharts have null values support. See this https://apexcharts.com/javascript-chart-demos/area-charts/chart-with-missing-data-or-null-values/. So if I am not wrong, just inserting a single `null` in value and the timestamp and type being the same (meaning non-null), in the very start of a chain should solve this. The querier can then before passing to UI, make some tweaks plus there needs to be a bit of changes in our parsing way in graphs which can be a good amount of work, but still doable. I guess in UI, the place where we segregate a single response into multiple data values for different graphs can be tweaked in a way that, as it sees a `"null"`, it inserts  `"null"` into those multiple data values and then this should be good to go.

Reply:
@Harkishen-Singh I just want to make sure that I have understood things correctly before making the next commit. 

1. As you said, the `addNullBlock` method along with some other methods will be added in the `tsdb` instead of the `main.go` file at the time when the chain starts.

2. The `zero` blocks that have been added will be changed to `null` blocks since the `apex charts` provides us a way to handle the `null` value as you mentioned. 

3. This zero block `0|0|0|0|0|0|0|0|0|0|0|0|0|0` is to be replaced with a single `null` and a `timestamp`, and thereby making the necessary changes in the `parser` and `UI` to handle this change.

Please correct me I am wrong or missed out on something. Thank you :)

Reply:
That looks correct